wayland-client/src/event_queue.rs

```diff
@@ -730,3 +730,32 @@ macro_rules! delegate_dispatch {
         )*
     };
 }
+
+/// A helper macro which delegates a set of [`Dispatch`] implementations for proxies to a no-op handler.
+///
+/// # Usage
+///
+/// This macro is useful to implement [`Dispatch`] for interfaces where events are unimportant to
+/// the current application and can be ignored.
+///
+/// ```
+/// use wayland_client::{delegate_noop, protocol::wl_subcompositor};
+/// delegate_noop!(ExampleApp: [wl_subcompositor::WlSubcompositor]);
+/// ```
+#[macro_export]
+macro_rules! delegate_noop {
+    ($dispatch_from: ty: [ $($interface: ty),* $(,)?]) => {
```

Comment:
You will need to copy the code from `delegate_dispatch` to also handle generics.

Reply:
Actually scratch this, the pull request for that is still pending